core/memory.go

```diff
@@ -165,8 +179,32 @@ func (memory *Memory) SetByteBank(value uint8, K uint8, offset uint16) {
 	case ioRegisterRegion:
 		memory.io[offset].Write(value)
 	case wramRegion:
-		memory.wram[(uint32(K)-0x7E)<<16+uint32(offset)] = value
+		memory.wram[(uint32(K%0x80)-0x7E)<<16+uint32(offset)] = value
 	case sramRegion:
-		memory.sram[offset] = value
+		memory.sram[memory.sm.getAddr(K, offset)] = value
 	}
 }
+
+type sramMapper struct {
```

Comment:
nit: can we use an interface here instead ? to avoid having the global variables `loromSramMapper` and `hiromSramMapper`

Reply:
Mmh I don't think an interface would fit there, what changes is not the implementation but only the parameters. I can add constructor for each if you prefer